km61/Raievskyi_Yevhenii/laboratory1.sql

```diff
@@ -1,2 +1,127 @@
--- LABORATORY WORK 1
--- BY Raievskyi_Yevhenii
+/*---------------------------------------------------------------------------
+1.Створити схему даних з назвою – прізвище студента, у якій користувач може: 
+створювати таблиці
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+
+CREATE USER raevskiy IDENTIFIED BY 1344
+DEFAULT TABLESPACE "USER"
+TEMPORARY TABLESPACE "TEMP";
+GRANT 'CONNECT' TO raevskiy;
+GRANT CREATE ANY TABLE TO raevskiy;
+
+
+/*---------------------------------------------------------------------------
+2. Створити таблиці, у яких визначити поля та типи. Головні та зовнішні ключі 
+створювати окремо від таблиць використовуючи команди ALTER TABLE. 
+Студент має мобільний номер іноземного оператора.
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+
+CREATE TABLE students
+(
+    stud_name VARCHAR(50)
+);
+
+ALTER TABLE students
+    ADD CONSTRAINT stud_pk PRIMARY KEY (stud_name);
+
+CREATE TABLE numbers
+(
+    phone_number NUMBER(15, 0),
+    phone_country VARCHAR(50)
+);
+
+ALTER TABLE numbers
+    ADD CONSTRAINT numbers_pk PRIMARY KEY (phone_number);
+    
+CREATE TABLE students_numbers
+(
+    stud_name_fk VARCHAR(50),
+    phone_number_fk NUMBER(15, 0)
+);
+
+ALTER TABLE students_numbers
+    ADD CONSTRAINT students_numbers_pk PRIMARY KEY (stud_name_fk, phone_number_fk);
+    
+ALTER TABLE students_numbers
+    ADD CONSTRAINT stud_name_fk FOREIGN KEY (stud_name_fk) REFERENCES students (stud_name);
+    
+ALTER TABLE students_numbers
+    ADD CONSTRAINT phone_number_fk FOREIGN KEY (phone_number_fk) REFERENCES numbers (phone_number);
+  
+/* --------------------------------------------------------------------------- 
+3. Надати додаткові права користувачеві (створеному у пункті № 1) для створення таблиць, 
+внесення даних у таблиці та виконання вибірок використовуючи команду ALTER/GRANT. 
+Згенерувати базу даних використовуючи код з теки OracleScript та виконати запити: 
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+
+GRANT SELECT ANY TABLE to raevskiy;
+GRANT INSERT ANY TABLE to raevskiy;
+
+
+/*---------------------------------------------------------------------------
+3.a. 
+Як звуть покупця, що купив найдешевший товар.
+Виконати завдання в Алгебрі Кодда. 
+4 бали
+---------------------------------------------------------------------------*/
+
+--Код відповідь:
+
+
+PROJECT 
```

Comment:
__Algebra Code the main task.
x=(Customers times Orders) where(Customers.cust_id =Orders.cust_id)
y=(x times OrderItems) where (Orders.order_num =OrderItems.order_num)
z=(y times Products) where (OrderItems.prod_id =Products.prod_id)
c=z.prod_price in (z project MIN(z.prod_price))   
b=z where c
a=b project cust_name

___SQL adittional task
select cust_name 
from Customers,Orders,OrderItems,Products
where (Customers.cust_id =Orders.cust_id and Orders.order_num =OrderItems.order_num
and OrderItems.prod_id =Products.prod_id and Products.prod_price in (select MIN(prod_price) from Products)